source/Support/Thread.cpp

```diff
@@ -57,17 +57,23 @@ getThreadCount() const noexcept
 
 void
 ThreadPool::
-async(
-    std::function<void(void)> f)
+wait()
 {
-    impl_->async(std::move(f));
+    impl_->wait();
 }
 
 void
 ThreadPool::
-wait()
+post(
+    any_callable<void(void)> f)
 {
-    impl_->wait();
+    auto sp = std::make_shared<
+        any_callable<void(void)>>(std::move(f));
```

Comment:
Can't be moved?

Reply:
Nope. We have to reimplement `llvm::ThreadPool` to allow moving. Until then, this is the acceptable hack.